views/Graph.react.js

```diff
@@ -390,54 +483,94 @@ class Graph extends Component {
     socket.close()
     window.removeEventListener('resize', this.handleResize)
   }
-
+  updateUserPreferences = (configuration) => {
+    let uid = this.props.user.uid
+    const createNewPreference = preparePreferences(
+      configuration,
+      this.state.preferences
+    )
+    return window
+      .fetch(`${basePath}/api/v1/users/${uid}/preferences`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        credentials: 'same-origin',
+        body: JSON.stringify({
+          preferences: createNewPreference,
+        }),
+      })
+      .then((res) => {
+        if (res.status === 201) {
+          window.location.replace(
+            routes.subjectView(
+              this.state.subject.project,
+              this.state.subject.sid
+            )
+          )
+        }
```

Comment:
should this call a prop that can redirect at the page level? Or are we at the page level?

Reply:
This is the page level of this component.

Reply:
yeah I think that makes sense.